packages/contracts-bedrock/src/safe/DeputyPauseModule.sol

```diff
@@ -0,0 +1,221 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Safe
+import { GnosisSafe as Safe } from "safe-contracts/GnosisSafe.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+
+// Contracts
+import { EIP712 } from "@solady/utils/EIP712.sol";
+
+// Libraries
+import { ECDSA } from "@solady/utils/ECDSA.sol";
+
+// Interfaces
+import { ISemver } from "src/universal/interfaces/ISemver.sol";
+import { IDeputyGuardianModule } from "src/safe/interfaces/IDeputyGuardianModule.sol";
+
+/// @title DeputyPauseModule
+/// @notice Safe Module designed to be installed in the Foundation Safe which allows a specific
+///         deputy address to act as the Foundation Safe for the sake of triggering the
+///         Superchain-wide pause functionality. Significantly simplifies the process of triggering
+///         a Superchain-wide pause without changing the existing security model.
+contract DeputyPauseModule is ISemver, EIP712 {
+    /// @notice Error message for invalid deputy.
+    error DeputyPauseModule_InvalidDeputy();
```

Comment:
Do we have semgrep rules for enforcing this convention?

Reply:
I don't believe we have that rule yet, but this was the convention we agreed upon [here](https://github.com/ethereum-optimism/design-docs/blob/main/solidity/revert-strings-to-custom-error-migration.md).

Reply:
Do we have an issue that tracks semgrep rules to add? We have https://github.com/ethereum-optimism/client-pod/issues/846 but unsure if that's the latest one you're using. Not a blocker to add the semgrep rule on this PR but would be nice for this to be enforced by semgrep

Reply:
Ok left a comment in 846.